web/routes/configPart.js

```diff
@@ -9,12 +9,12 @@ const parts = {
 
 const gekkoRoot = __dirname + '/../../';
 
-module.exports = function *() {
-  if(!_.has(parts, this.params.part))
-    return this.body = 'error :(';
+module.exports = function (ctx) {
+  if(!_.has(parts, ctx.params.part))
+    return ctx.body = 'error :(';
 
-  const fileName = gekkoRoot + '/' + parts[this.params.part] + '.toml';
-  this.body = {
-    part: yield fs.readFile(fileName, 'utf8') 
+  const fileName = gekkoRoot + '/' + parts[ctx.params.part] + '.toml';
+  ctx.body = {
+    part: fs.readFileSync(fileName, 'utf8')
```

Comment:
Would you mind not doing sync calls in the API? Those block everything else (ws message passing, ws heartbeats, other API calls, etc)!

Reply:
ok, I will refactor with promisify async/await.